ballerina/hub_client.bal

```diff
@@ -46,67 +45,53 @@ public client class HubClient {
         self.hub = subscription.hub;
         self.topic = subscription.hubTopic;
-        self.linkHeaderValue = generateLinkUrl(self.hub,  self.topic);
-        self.secret = subscription?.hubSecret is string ? <string>subscription?.hubSecret : "";
+        self.hubLinks = generateLinkUrl(self.hub,  self.topic);
+        self.secret = subscription?.hubSecret;
         self.httpClient = check retrieveHttpClient(subscription.hubCallback, retrieveHttpClientConfig(config));
     }
 
     # Distributes the published content to the subscribers.
     # ```ballerina
-    # ContentDistributionSuccess|SubscriptionDeletedError|error? publishUpdate = websubHubClientEP->notifyContentDistribution(
-    #   { 
-    #       content: "This is sample content" 
-    #   }
-    # );
+    # ContentDistributionSuccess publishUpdate = check websubHubClientEP->notifyContentDistribution({ content: "This is sample content" });
     # ```
     #
     # + msg - Content to be distributed to the topic subscriber 
     # + return - An `websubhub:Error` if an exception occurred, a `websubhub:SubscriptionDeletedError` if the subscriber responded with `HTTP 410`,
     #            or else a `websubhub:ContentDistributionSuccess` for successful content delivery
-    isolated remote function notifyContentDistribution(ContentDistributionMessage msg) 
+    isolated remote function notifyContentDistribution(ContentDistributionMessage message) 
                                 returns ContentDistributionSuccess|SubscriptionDeletedError|Error {
-        string contentType = retrieveContentType(msg.contentType, msg.content);
         http:Request request = new;
-        string queryString = "";
-        match contentType {
-            mime:APPLICATION_FORM_URLENCODED => {
-                map<string> messageBody = <map<string>> msg.content;
-                queryString += retrieveTextPayloadForFormUrlEncodedMessage(messageBody);
-                request.setTextPayload(queryString, mime:APPLICATION_FORM_URLENCODED);
-            }
-            _ => {
-                request.setPayload(msg.content);
-            }
-        }
-
-        if msg?.headers is map<string|string[]> {
-            var headers = <map<string|string[]>>msg?.headers;
+        map<string|string[]>? headers = message?.headers;
+        if headers is map<string|string[]> {
             foreach var [header, values] in headers.entries() {
                 if values is string {
                     request.addHeader(header, values);
                 } else {
-                    foreach var value in <string[]>values {
+                    foreach string value in values {
                         request.addHeader(header, value);
                     }
                 }
             }
         }
+        string contentType = retrieveContentType(message.contentType, message.content);
+        json|xml|byte[] payload = retrieveRequestPayload(contentType, message.content);
+        request.setPayload(payload);
```

Comment:
I assume the payload can contain the string type also. if so add it through a json type can make the content type `application/json` instead of being `text/plain`. Is it fine for the usecase?

Reply:
In the `http:Request` we are manually setting the `content-type`.